CameHeader value is wrongly interpreted as BigDecimal and causes ClassCastException

using `camel-kafka-connector 0.9.0`
using sink connector `CamelAws2s3SinkConnector`

Kafka header `CamelHeader.Id: l1kt74gl9w00` is logged as:
`ConnectHeader(key=CamelHeader.Id, value= l1kt74gl9w00, schema=Schema{STRING})`

Kafka header `CamelHeader.Id: 69203120e780` is logged as:
`ConnectHeader(key=CamelHeader.Id, value=6.9203120E+787, schema=Schema{org.apache.kafka.connect.data.Decimal:BYTES})`

causing the task to fail:

```
org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:614)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:329)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:232)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:201)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:189)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:238)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassCastException
```
```
Caused by: java.lang.ClassCastException: class java.math.BigDecimal cannot be cast to class [B (java.math.BigDecimal and [B are in module java.base of loader 'bootstrap')
	at org.apache.camel.kafkaconnector.CamelSinkTask.mapHeader(CamelSinkTask.java:233)
	at org.apache.camel.kafkaconnector.CamelSinkTask.put(CamelSinkTask.java:184)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:586)
	... 10 more
```

A workaround or any suggestion appreciated.

--- a/core/src/main/java/org/apache/camel/kafkaconnector/CamelSinkTask.java
+++ b/core/src/main/java/org/apache/camel/kafkaconnector/CamelSinkTask.java
@@ -240,7 +240,9 @@ public class CamelSinkTask extends SinkTask {
         final String key = StringHelper.after(header.key(), prefix, header.key());
         final Schema schema = header.schema();
 
-        if (schema.type().equals(Schema.BYTES_SCHEMA.type()) && Objects.equals(schema.name(), Decimal.LOGICAL_NAME)) {
+        if (schema.type().equals(Schema.BYTES_SCHEMA.type())
+                && Objects.equals(schema.name(), Decimal.LOGICAL_NAME)
+                && header.value() instanceof byte[]) {
             destination.put(key, Decimal.toLogical(schema, (byte[]) header.value()));
         } else {
             destination.put(key, header.value());

--- a/core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java
+++ b/core/src/main/java/org/apache/camel/kafkaconnector/CamelSourceTask.java
@@ -36,7 +36,6 @@ import org.apache.camel.kafkaconnector.utils.SchemaHelper;
 import org.apache.camel.kafkaconnector.utils.TaskHelper;
 import org.apache.camel.support.UnitOfWorkHelper;
 import org.apache.kafka.clients.producer.RecordMetadata;
-import org.apache.kafka.connect.data.Decimal;
 import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.errors.ConnectException;
 import org.apache.kafka.connect.source.SourceRecord;
@@ -357,8 +356,9 @@ public class CamelSourceTask extends SourceTask {
             } else if (value instanceof Date) {
                 record.headers().addTimestamp(keyCamelHeader, (Date)value);
             } else if (value instanceof BigDecimal) {
-                Schema schema = Decimal.schema(((BigDecimal)value).scale());
-                record.headers().add(keyCamelHeader, Decimal.fromLogical(schema, (BigDecimal)value), schema);
+                //XXX: kafka connect configured header converter takes care of the encoding,
+                //default: org.apache.kafka.connect.storage.SimpleHeaderConverter
+                record.headers().addDecimal(keyCamelHeader, (BigDecimal)value);
             } else if (value instanceof Double) {
                 record.headers().addDouble(keyCamelHeader, (double)value);
             } else if (value instanceof Float) {

--- a/core/src/test/java/org/apache/camel/kafkaconnector/CamelSinkTaskTest.java
+++ b/core/src/test/java/org/apache/camel/kafkaconnector/CamelSinkTaskTest.java
@@ -33,6 +33,7 @@ import org.apache.kafka.connect.data.SchemaAndValue;
 import org.apache.kafka.connect.data.SchemaBuilder;
 import org.apache.kafka.connect.errors.ConnectException;
 import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.storage.SimpleHeaderConverter;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -118,7 +119,13 @@ public class CamelSinkTaskTest {
         int myInteger = 100;
         Long myLong = new Long("100");
         BigDecimal myBigDecimal = new BigDecimal(1234567890);
-        Schema schema = Decimal.schema(myBigDecimal.scale());
+        Schema myBigDecimalSchema = Decimal.schema(myBigDecimal.scale());
+        //reproducing bigDecimal encoding by kafka connect
+        BigDecimal kafkaBigDecimal = new BigDecimal("6.9203120E+787");
+        Schema kafkaBigDecimalSchema = Decimal.schema(kafkaBigDecimal.scale());
+        SimpleHeaderConverter shc = new SimpleHeaderConverter();
+        byte[] persistedBytes = shc.fromConnectHeader("", "MyBigDecimal", kafkaBigDecimalSchema, kafkaBigDecimal);
+        SchemaAndValue sav = shc.toConnectHeader("", "MyBigDecimal", persistedBytes);
 
         List<SinkRecord> records = new ArrayList<SinkRecord>();
         SinkRecord record = new SinkRecord(TOPIC_NAME, 1, null, "test", null, "camel", 42);
@@ -129,7 +136,9 @@ public class CamelSinkTaskTest {
         record.headers().addDouble(CamelSinkTask.HEADER_CAMEL_PREFIX + "MyDouble", myDouble);
         record.headers().addInt(CamelSinkTask.HEADER_CAMEL_PREFIX + "MyInteger", myInteger);
         record.headers().addLong(CamelSinkTask.HEADER_CAMEL_PREFIX + "MyLong", myLong);
-        record.headers().add(CamelSinkTask.HEADER_CAMEL_PREFIX + "MyBigDecimal", Decimal.fromLogical(schema, myBigDecimal), schema);
+        record.headers().add(CamelSinkTask.HEADER_CAMEL_PREFIX + "MyBigDecimal", Decimal.fromLogical(myBigDecimalSchema, myBigDecimal), myBigDecimalSchema);
+        record.headers().add(CamelSinkTask.HEADER_CAMEL_PREFIX + "KafkaBigDecimal", sav.value(), sav.schema());
+
         records.add(record);
         sinkTask.put(records);
 
@@ -145,6 +154,7 @@ public class CamelSinkTaskTest {
         assertEquals(myInteger, exchange.getIn().getHeader("MyInteger"));
         assertEquals(myLong, exchange.getIn().getHeader("MyLong", Long.class));
         assertEquals(myBigDecimal, exchange.getIn().getHeader("MyBigDecimal", BigDecimal.class));
+        assertEquals(kafkaBigDecimal, exchange.getIn().getHeader("KafkaBigDecimal", BigDecimal.class));
 
         sinkTask.stop();
     }

--- a/core/src/test/java/org/apache/camel/kafkaconnector/CamelSourceTaskTest.java
+++ b/core/src/test/java/org/apache/camel/kafkaconnector/CamelSourceTaskTest.java
@@ -312,7 +312,7 @@ public class CamelSourceTaskTest {
         List<SourceRecord> results = sourceTask.poll();
         assertEquals(1, results.size());
         Header bigDecimalHeader = results.get(0).headers().allWithName(CamelSourceTask.HEADER_CAMEL_PREFIX + "bigdecimal").next();
-        assertEquals("[B", bigDecimalHeader.value().getClass().getName());
+        assertTrue(bigDecimalHeader.value() instanceof BigDecimal);
         assertEquals(Decimal.class.getName(), bigDecimalHeader.schema().name());
         assertEquals(Schema.Type.BYTES, bigDecimalHeader.schema().type());
 

